components/download/combined-install.tsx

```
@@ -0,0 +1,577 @@
+    try {
+      const result = await invoke<{ id: string; version: string }[]>("get_fabric_api_versions", { mcVersion: version.id });
+      setFabricApiVersions(result.map((v) => ({ id: v.id, version: v.version, releaseDate: "" })));
+    } catch (err) {
+      console.error("获取Fabric API版本列表失败:", err);
+    } finally {
+      setLoadingFabricApi(false);
+    }
+  }, [version.id, fabricApiVersions.length, loadingFabricApi]);
+
+  useEffect(() => {
+    if (hasFabric) void fetchFabricApiVersions();
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/components/download/combined-install.tsx:142:25
  140 |
  141 |   useEffect(() => {
> 142 |     if (hasFabric) void fetchFabricApiVersions();
      |                         ^^^^^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  143 |   }, [hasFabric, fetchFabricApiVersions]);
  144 |
  145 |   const openFabricApiView = () => {

(react-hooks/set-state-in-effect)